lib/library.rb

```diff
@@ -0,0 +1,57 @@
+class Library
+  attr_reader :name, :books, :authors, :checked_out_books
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+  end
+
+  def add_author(author_obj)
+    @authors << author_obj
+  end
+
+  def add_book(book_obj)
+    @books << book_obj
+  end
+
+  def publication_time_frame_for(author_obj)
+    authors_books = @books.find_all do |book|
```

Comment:
No need to go through all the books. We have the author, and we can get all this author's books by calling `author_obj.books` :)